### ISSUE TITLE
Output issues with neofetch 7.1.0 on Fedora

### ISSUE THREAD

**Issue by Skaldebane** [dylanaraps/neofetch#1952]
## Description

Hi! Today I updated neofetch (through dnf), I like the new icon, but...
![Screenshot from 2021-11-10 07-08-20](https://user-images.githubusercontent.com/35589610/141059827-4de474cc-329b-4af3-9a50-5073f2719c7b.png)
What's all this spacing here?

(cmd: `neofetch --speed_shorthand on --cpu_temp C --memory_display infobar --disable packages`)

#### Neofetch version

This is neofetch 7.1.0.
But again, why do I get these errors when requesting the version?
![Screenshot from 2021-11-10 07-16-51](https://user-images.githubusercontent.com/35589610/141060264-2a1aecc3-26c9-4b35-a5d7-8ce41f0dac25.png)

## Config file

[config.conf file](https://pastebin.ubuntu.com/p/khpdc84RKk/)

## Verbose log

[neofetchlog file](https://pastebin.ubuntu.com/p/Qqy593xMKz/)

**Comment by Skaldebane** [comment]
This also seems to happen differently if I run the same command normally (note that the cmd above is being run in .bashrc):
![Screenshot from 2021-11-10 07-31-08](https://user-images.githubusercontent.com/35589610/141062011-80cab169-e22a-4673-ac96-436407cf3730.png)

**Comment by dylanaraps** [comment]
The errors are from running in verbose mode. These are normal.
Does the issue occur in master?

**Comment by dylanaraps** [comment]
I cannot reproduce using the provided commands.

**Comment by Skaldebane** [comment]
I'll try the master version and reply right away.

And also note that this doesn't happen when I run that command normally, only when it's run through .bashrc.

**Comment by Skaldebane** [comment]
I installed the master version in a `toolbox` Fedora container, still the same issue.
It shows this "istrom" inside the memory indicator `infobar`, plus some line breaks (the line breaks count depends on the situation: .bashrc > normal > verbose (=0)).

Main version downloaded from dnf (through .bashrc):
![Screenshot from 2021-11-10 07-08-20](https://user-images.githubusercontent.com/35589610/141611911-37f1113b-f0ef-4975-8b75-5f27bc5a182e.png)
(Notice how the line breaks try to fit the terminal height?)

-------
Master version, inside a toolbox container:
![Screenshot from 2021-11-13 09-28-05](https://user-images.githubusercontent.com/35589610/141611923-083037a5-0754-4cc1-80af-11d843e684c1.png)

-------
Master version, inside a toolbox container, verbose mode:
![Screenshot from 2021-11-13 09-28-40](https://user-images.githubusercontent.com/35589610/141611956-1e69c6cd-dcf6-4362-bb65-9624bdf2f305.png)

**Comment by Skaldebane** [comment]
Maybe this has something to do with some external dependency?

**Comment by dylanaraps** [comment]
Does the issue occur in a different terminal emulator?

**Comment by Skaldebane** [comment]
Tried it on XTerm, same thing:
![Screenshot from 2021-11-14 09-40-43](https://user-images.githubusercontent.com/35589610/141673929-c4122f70-6f9d-4b0b-b600-98a3480a98e7.png)

**Comment by skaldebane** [comment]
I played around with the shell commands inside neofetch and managed to replicate the bar with the same code from `neofetch`, and it works without problems... Seems like something external adds these characters and line breaks in there? Or interprets something?

It's such a weird bug, as the code seems to be fine, and it always worked on all distros I've ever used, including this exact system before the last update. The 'bar' (bar, infobar, barinfo) memory indicators seem to break something in there.

**Comment by skaldebane** [comment]
Any updates on this topic?

**Comment by dylanaraps** [comment]
Yes. I managed to reproduce this issue. Have not had the time to fix it though.

**Comment by thelfensdrfer** [comment]
I am having the same issue on rocky linux 8.5. Happy to test/debug the issue with a little pointer in the right direction.

**Comment by Crinisus** [comment]
I can confirm it happens in both Fedora 35 and 36 Prerelease as well.

**Comment by skaldebane** [comment]
I did some debugging, and it seems `bar_color_total` is where this issue stems from.

Debugging shows this value given to `bar_color_total`:

![image](https://user-images.githubusercontent.com/35589610/160493099-b817a26b-2ae2-4510-9f4d-e1f48c008dcd.png)

Value: "[38;5;distrom"

The first tofu character seems to be the one filling all the space in the terminal.

Weirdly enough, this is coming from the 3rd argument, which is coming from `get_args()` after the option `--bar_colors`. We're not even setting it anywhere!

I don't really code in bash, and only have some basic knowledge as any regular Linux user, so I don't really know how to fix this problem. I hope you can find a solution soon 🙏️

**Comment by skaldebane** [comment]
## Actually, there's a workaround: Simply add `--bar_colors` without any values to the end of your neofetch command.
This will overwrite the broken value that's automatically added.

Example: `neofetch --disable packages --memory_display infobar --bar_colors`

![image](https://user-images.githubusercontent.com/35589610/160494795-b8a2b1e4-b8b6-4d9d-83e1-c440e4362c46.png)

**Comment by thelfensdrfer** [comment]
You are right. A temporary fix for me was to change the config from:

```
bar_color_elapsed="distro"
bar_color_total="distro"
```

to

```
# 3 and 4 can be any numbers/colors
bar_color_elapsed="3"
bar_color_total="4"
```

**Comment by skaldebane** [comment]
So this problem is from this code here:

![image](https://user-images.githubusercontent.com/35589610/160497944-a0976170-53ca-4651-9932-a2b8b57902fe.png)

Which works just fine for `bar_color_elapsed`, but not for `bar_color_total`.

Notice the last line? If the `bar_color_total` doesn't match the previous two cases, it just passes its current value as it is (which is "distro") to the `color()` function, which in turn just inserts it inside `printf`.
